packages/indexer/src/indexer.ts

```diff
@@ -31,8 +31,13 @@ export interface IndexerHooks<TFilter, TBlock, TRet> {
     options: StreamDataOptions;
   }) => void;
   "connect:after": () => void;
-  "handler:before": ({ block }: { block: TBlock }) => void;
+  "handler:before": ({
+    block,
+    finality,
+    endCursor,
+  }: { block: TBlock; finality: DataFinality; endCursor?: Cursor }) => void;
   "handler:after": ({ output }: { output: TRet[] }) => void;
+  "handler:exception": ({ error }: { error: Error }) => void;
```

Comment:
Is this handler called anywhere? I think we should catch the handler exception and then rethrow it after calling the hook.